PR ajustes REST para branch develop

LucasDelRio7 opened the pull request:
Ajuste no nome da Controller + Rota GET com query string para DDD  (padrão REST);
Criada class Utils para remover validações da Service;

Commit:
Ajuste no nome da Controller + Rota GET com query string para DDD  (padrão REST);
Criada class Utils para remover validações da Service;

Commit:
Ajustando teste Created
Retornando erros concatenados e de forma única

raphaelrogenski commented on src/Contact.Api/Controllers/ContactController.cs:
```diff
@@ -4,8 +4,8 @@
 
 namespace Contacts.Api.Controllers
 {
-    [Route("api/[controller]")]
```
Não seria mais prático, utilizar a maneira que estava do api/[controller], que a própria API substitui pelo nome da controller, que no caso é Contact? Qualquer coisa se quiser manter no plural, seria só mudar para ContactsController o nome da classe...

LucasDelRio7 replied:
Seguindo o padrão de nomenclaturas REST não, o nome da Controller fica no singular e o "resource" no plural

raphaelrogenski commented on src/Contact.Application/Contacts/Services/ContactService.cs:
```diff
@@ -52,93 +47,59 @@ public void Delete(Guid id)
         var emailIsEmpty = string.IsNullOrWhiteSpace(vo.EmailAddress);
         if (emailIsEmpty)
-            throw new ArgumentException("Email Address shouldn't be empty!");
+            mensagemErro += "Email Address shouldn't be empty! \n";
 
         //CHECK IF DATA IS VALID
 
-        var phoneDDDIsInvalid = !ValidatePhoneDDD(vo.PhoneDDD);
+        var phoneDDDIsInvalid = !StringUtils.ValidatePhoneDDD(vo.PhoneDDD);
         if (phoneDDDIsInvalid)
-            throw new ArgumentException("Phone DDD is invalid!");
+            mensagemErro += "Phone DDD is invalid! \n";
 
-        var phoneNumberIsInvalid = !ValidatePhoneNumber(vo.PhoneNumber);
+        var phoneNumberIsInvalid = !StringUtils.ValidatePhoneNumber(vo.PhoneNumber);
         if (phoneNumberIsInvalid)
-            throw new ArgumentException("Phone Number is invalid!");
+            mensagemErro += "Phone Number is invalid! \n";
 
-        var emailIsInvalid = !ValidateEmailAddress(vo.EmailAddress);
+        var emailIsInvalid = !StringUtils.ValidateEmailAddress(vo.EmailAddress);
         if (emailIsInvalid)
-            throw new ArgumentException("Email Address is invalid!");
+            mensagemErro += "Email Address is invalid! \n";
 
         //CHECK IF DATA IS ALREADY IN USE
 
         var nameAlreadyInUse = Repository.ContactNameAlreadyExists(vo.Name, vo.Id);
         if (nameAlreadyInUse)
-            throw new ArgumentException("Name already in use!");
+            mensagemErro += "Name already in use! \n";
 
         var phoneAlreadyInUse = Repository.ContactPhoneAlreadyExists(vo.PhoneDDD, vo.PhoneNumber, vo.Id);
         if (phoneAlreadyInUse)
-            throw new ArgumentException("Phone already in use!");
+            mensagemErro += "Phone already in use! \n";
 
         var emailAlreadyInUse = Repository.ContactEmailAlreadyExists(vo.EmailAddress, vo.Id);
         if (emailAlreadyInUse)
-            throw new ArgumentException("Email already in use!");
```
Duas observações sobre essa mudança de retornar todos as falhas de validações ao invés de só a primeira:
1) Se receber o email vazio, ele vai verificar se o email está vazio, e também vai verificar se o regex bate. Talvez daria pra remover a validação do IsNullOrWhitespace, pra não realizar a mesma verificação duas vezes.
2) Se o nome, telefone ou email estiverem inválidos, o ideal seria que nem verificasse no banco de dados se ele já existe ou não, pois fará uma consulta a toa, visto que não irá cadastrar.

LucasDelRio7 replied:
Vou rever a lógica, eu só troquei mesmo a ideia, mas faz sentido esse pontos ai

raphaelrogenski commented on src/Contact.Application/Contacts/Services/ContactService.cs:
```diff
@@ -52,93 +47,59 @@ public void Delete(Guid id)
 
     private void EnsureValidation(ContactVO vo)
     {
+        string mensagemErro = string.Empty;
```
Como está tudo em inglês, seria melhor traduzir essa variável pro inglês tbm...

LucasDelRio7 replied:
Feito

raphaelrogenski commented on src/Contact.Infrastructure/Services/ServiceBase.cs:
Aparentemente não mudou nada nesse arquivo, parece ser algo só de encoding ou line ending.

raphaelrogenski commented on src/Contact.Test/Tests/ContactControllerTest.cs:
Os testes falharam, pois está sendo usado Assert.Equals...
Com a mudança para retornar os multiplos erros, dá pra trocar pelo Assert.Contains([expected], badRequestResult.Value.ToString());

LucasDelRio7 replied:
Corrigido

raphaelrogenski commented on src/Contact.Api/Program.cs:
No ApplicationTest, usar o Api.Application.Run([]), ao invés de Api.Program.Main([]);

LucasDelRio7 replied:
Feito, alterei tbm o método para async Task

Commit:
ajustes nas validações e testes

raphaelrogenski commented on src/Contact.Test/Tests/ApplicationTest.cs:
Da maneira que está, com o await está rodando a aplicação e fica nessa linha eternamente, dessa forma os testes nunca finalizam...

raphaelrogenski replied:
Só rodar os testes via visual studio que você já consegue ver se deu boa as alterações...

raphaelrogenski committed:
ContactService: Keeping the validations in a simple way.
ContactControllerTest: Fixes due to ContactService changes.
RepositoryBase: Added Id not found verification.

raphaelrogenski committed:
Application.cs, Program.cs, changed the way to run the app as webapplication or tests.

raphaelrogenski reviewed (approved):
Tudo certo...

raphaelrogenski committed:
updating sonar exclusions to Program.cs and Migrations folder (auto-generated)
